### PR TITLE
Phishing indicators hunting

### PR DESCRIPTION
**Pull request by tomer-pan**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://jira-hq.paloaltonetworks.local/browse/CIAC-5148

## Description
Adding Email Indicators hunting branch to the Phishing v3 playbook.

## Minimum version of Cortex XSOAR
- [x] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

### CODE REVIEW

**Review by idovandijk — COMMENTED** [review]
Amazing job Tomer. See my comments below. I will number them for easier reference .

1. Small typo in playbook input
image

2. I suggest to add ignore-case:
image

3. Note that you are actually passing all the DBotScores regardless of whether they're malicious or not. I know that inside it you are passing only the malicious ones, but I suggest to pass only the malicious ones to begin with, for the same of performance (usually there will be many DBotScores, and we are creating a copy of them inside the subplaybook without using many of them).
image

4. Since you are no longer hunting for clicks only from the same email, do we still need the MessageID input?
image

5. Question - is it possible that in the Microsoft DB there will be events where `RecipientEmailAddress` is empty, which will sabotage the search when we have no ListenerMailbox (such as when we open incident manually)? Or will there always be a value? You can quickly validate by searching all events where `RecipientEmailAddress` is empty.
image

6. Not sure if this is there on purpose or a mistake:
image

7. I'm confused about this part. The `UrlDomain` in the `EmailUrlInfo` table should hold "Domain name or host name of the URL" according to the documentation. But this query checks if `UrlDomain` contains the `IPAddress` input:
image

Also confused about the 2 separate searches searching 2 different tables for the IPAddresses, we should explain in the playbook why separate searches are done.

8. Question regarding the saved results:
image
I think that task 16 is already collecting all the events not found in task 10, so why do we need task 10?

9. Suggestion - you can merge these into 1 task: 
image

You can set the `Microsoft365Defender.RetrievedEmails` key and append the `RetrievedEmailsJoined` where `RetrievedEmailsJoined.NetworkMessageId` not in `
Microsoft365Defender.RetrievedEmails.NetworkMessageId`

10. In the `Phishing - Handle Microsoft 365 Defender Results` playbook, this comparison is problematic because sometimes the value is a comma separated string:
image
image

11. The `Phishing - Search Related Incidents (Defender 365)` playbook runs and creates the `EmailFilesRetrieval` key, which should have email files of emails that don't have an incident, filtered by the `reportedinternetmessageid` of the current incident, but it can get there even if there is no `reportedinternetmessageid`. What will happen in that case? Can it cause a recursion of incidents that are duplicate of the current one?

12. For the layout I suggest to explain that `Total Malicious URLs Clicks` includes both blocked, allowed and clicked through events (not sure if it's self explanatory).

To sum up - excellent job, I could see that you thought of many edge cases here and did your best to explain the flow. In my humble opinion I think the most important thing about this PR is doing our best to help the least experienced user understand exactly what's happening. I don't mind joining the technical review process with you as well and making sure everything is explained thoroughly, as I had (even after numerous calls together) a hard time figuring out some parts.

**Comment by tomer-pan** [comment]
Thanks for the review @idovandijk ! Please see my answers below:
1) Will do
2) Will do
3) Will do
4) It is necessary for cases this playbook would run in a different context, when someone would want to get the clicks inside a specific email
5) This is okay - I tested it
6) Seems like a mistake - nice catch
7) This is when the IP is part of the url - it will be part of the UrlDomain field. Also as for the 2 separate searches - I will add a description to the tasks which explains the separation.
8) Since some of the results can show up several times, some with extra data (like attachments or url data) and some without it, we would want to save the objects with the extra data first (task 10) and then add to it the objects without the extra data in task 16 (filtering out the objects already in the list created in task 10).
9) I'm not sure that's possible, let's sync on that.
10) This is also works on comma separated lists, I tried it and it works (as you can see in the "BCCs" incident).
11) I'm not sure I got this as well - In case there's no "reportedemailmessageid" in one of the incidents which represents one of the found emails we can't know for sure that the incident actually represents the actual email. In that case there will be no recursion what so ever since if the customer choose to create a new incident for it, it will be created with the relevant reported email message id, and of not, then no recursion is relevant. Please let me know if we need to go over that to.
12) I will add the description.

File: Packs/Microsoft365Defender/Playbooks/playbook-Microsoft_365_Defender_-_Get_Email_URL_Clicks.yml [hunk 1]
```diff
@@ -622,4 +627,4 @@ outputs:
   type: unknown
 tests:
 - No tests (auto formatted)
-fromversion: 6.5.0
```
Comment (idovandijk) [inline Packs/Microsoft365Defender/Playbooks/playbook-Microsoft_365_Defender_-_Get_Email_URL_Clicks.yml:625]: Since `Emails Indicators Hunt` is from 6.2.0, this one must also be 6.2.0

**Comment by tomer-pan** [comment]
> The PR looks good, I just added a comment about `fromversion` of one of the playbooks. If that playbook cannot be used as standalone then maybe we should change the other playbooks to have `fromversion 6.2.0` instead of 6.5.0

This playbook can be used as a standalone playbook.
Also, for the new playbooks which are not released yet, the fromversion should be 6.8

File: Packs/Microsoft365Defender/Playbooks/playbook-Microsoft_365_Defender_-_Get_Email_URL_Clicks.yml (hunk 1 above)
Comment (tomer-pan) [reply Packs/Microsoft365Defender/Playbooks/playbook-Microsoft_365_Defender_-_Get_Email_URL_Clicks.yml:625]: This is a mistake, the new playbook should be from 6.8, will fix that.

File: Packs/Microsoft365Defender/Playbooks/playbook-Microsoft_365_Defender_-_Emails_Indicators_Hunt.yml [hunk 2]
```diff
@@ -535,16 +533,14 @@ tasks:
...
-      id: 6323c0a2-b979-4320-85be-3193ad8ee226
+      id: a9b33515-9725-4e37-8188-2120f8fb69e3
       version: -1
-      name: Retrieve emails data (IPAddress)
-      description: 'Advanced hunting is a threat-hunting tool that uses specially
-        constructed queries to examine the past 30 days of event data in Microsoft
-        365 Defender. Details on how to write queries: https://docs.microsoft.com/en-us/microsoft-365/security/defender/advanced-hunting-query-language?view=o365-worldwide.'
+      name: Retrieve emails data (IPAddress) - EmailEvents table
       script: Microsoft 365 Defender|||microsoft-365-defender-advanced-hunting
+      description: 'This task will retrieve emails data based on the EmailEvents table. It will look for the given IP in the "SenderIPv4" and in the "SenderIPv46" properties.'
```
Comment (ShirleyDenkberg) [inline Packs/Microsoft365Defender/Playbooks/playbook-Microsoft_365_Defender_-_Emails_Indicators_Hunt.yml:543]: Should this be:
...the given IP in the "SenderIPv4" and in the "**SenderIPv6**" properties.'

**Comment by ShirleyDenkberg** [comment]
@idovandijk @michalgold @ostolero Doc review completed.

**Review by idovandijk — APPROVED** [review]
